### ISSUE TITLE
Move File/Folder too slow/often fails. Disable additional checks.

### ISSUE THREAD

**Issue by jflifecycle** [microsoft/vscode#164172]
Type: <b>Performance Issue</b>

When moving files/folders in a project, VSCode will do additional checking based on settings (Like "Typescript: Update imports on file move") or extensions.

This causes a significant slowdown while vscode figures out what do about the files and is not necessarily proportional to the size of the project. It also depends on potentially a number of extensions not messing this up, during which time, you cannot do a number of operations.

This OFTEN fails and freezes access to the files/folder which can prevent them from being moved unless you close code.exe to release the files, and then move them in the file explorer and then reopen VScode.

I propose a setting that disables ANY additional checking of file moves, and just lets the move happen.

VS Code version: Code 1.72.2 (d045a5eda657f4d7b676dedbfa7aab8207f8a075, 2022-10-12T22:15:18.074Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz (8 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.75GB (2.03GB free)|
|Process Argv|--crash-reporter-id a07bbebd-683b-4bed-a1ff-d43eea21fcbf|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	    83	 23596	code main
    0	   271	 21996	   gpu-process
    0	    69	 22344	   shared-process
    0	    12	 37192	     fileWatcher
    0	    37	 37904	     ptyHost
    0	     1	 28200	       "C:\Program Files\Git\bin\bash.exe" --login
    0	     3	 15696	         "C:\Program Files\Git\bin\..\usr\bin\bash.exe" --login
    0	     3	 29464	       console-window-host (Windows internal process)
    0	     5	 39820	       "C:\Program Files\Git\bin\bash.exe" --login
    0	    11	 13672	         "C:\Program Files\Git\bin\..\usr\bin\bash.exe" --login
    0	     6	 45472	       console-window-host (Windows internal process)
    0	    24	 40920	     fileWatcher
    0	    66	 45468	     fileWatcher
    0	     6	 24932	   crashpad-handler
    0	   106	 25888	   extensionHost
    0	     7	 10816	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\html-language-features\server\dist\node\htmlServerMain" --node-ipc --clientProcessId=25888
    0	    10	 21352	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\SqlToolsResourceProviderService.exe --log-file c:\Users\jermaine.francoeur\AppData\Roaming\Code\logs\20221019T101957\exthost3\ms-mssql.mssql\resourceprovider.log --tracing-level Critical
    0	     2	 31540	       console-window-host (Windows internal process)
    0	    12	 23672	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0\server --logToConsole --ngProbeLocations c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas\private --includeAutomaticOptionalChainCompletions --includeCompletionsWithSnippetText --tsProbeLocations ,c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas\private --node-ipc --clientProcessId=25888
    0	    10	 33180	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\dbaeumer.vscode-eslint-2.2.6\server\out\eslintServer.js --node-ipc --clientProcessId=25888
    0	    12	 40652	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\MicrosoftSqlToolsServiceLayer.exe
    0	     2	 27460	       console-window-host (Windows internal process)
    0	   253	 29980	   window (index.ts - xcas-spa - Visual Studio Code)
    0	   117	 32032	   extensionHost
    0	    47	 17824	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --serverMode partialSemantic --useInferredProjectPerProjectRoot --disableAutomaticTypingAcquisition --cancellationPipeName C:\Users\JERMAI~1.FRA\AppData\Local\Temp\vscode-typescript\8e00a392fb96b676afcc\tscancellation-d0130e43f349b470c976.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	   211	 17960	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName C:\Users\JERMAI~1.FRA\AppData\Local\Temp\vscode-typescript\8e00a392fb96b676afcc\tscancellation-e9ccd069d061303be76e.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	    18	 41128	       "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:/Users/jermaine.francoeur/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typingsInstaller.js" --globalTypingsCacheLocation C:/Users/jermaine.francoeur/AppData/Local/Microsoft/TypeScript/4.8 --enableTelemetry --typesMapLocation "c:/Users/jermaine.francoeur/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typesMap.json" --validateDefaultNpmLocation
    0	    18	 18228	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\html-language-features\server\dist\node\htmlServerMain" --node-ipc --clientProcessId=32032
    0	    10	 28960	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\SqlToolsResourceProviderService.exe --log-file c:\Users\jermaine.francoeur\AppData\Roaming\Code\logs\20221019T101957\exthost1\ms-mssql.mssql\resourceprovider.log --tracing-level Critical
    0	     2	 23424	       console-window-host (Windows internal process)
    0	    17	 30068	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\css-language-features\server\dist\node\cssServerMain" --node-ipc --clientProcessId=32032
    0	    12	 36948	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\MicrosoftSqlToolsServiceLayer.exe
    0	     2	  8112	       console-window-host (Windows internal process)
    0	    17	 39368	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\dbaeumer.vscode-eslint-2.2.6\server\out\eslintServer.js --node-ipc --clientProcessId=32032
    0	   363	 48788	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0\server --logToConsole --ngProbeLocations c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas-spa --includeAutomaticOptionalChainCompletions --includeCompletionsWithSnippetText --tsProbeLocations ,c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas-spa --node-ipc --clientProcessId=32032
    0	    72	 51064	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=32032
    0	    85	 33648	   extensionHost
    0	     9	  3188	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\SqlToolsResourceProviderService.exe --log-file c:\Users\jermaine.francoeur\AppData\Roaming\Code\logs\20221019T101957\exthost2\ms-mssql.mssql\resourceprovider.log --tracing-level Critical
    0	     2	 23180	       console-window-host (Windows internal process)
    0	    16	  4772	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\html-language-features\server\dist\node\htmlServerMain" --node-ipc --clientProcessId=33648
    0	    19	 10796	     c:\Users\jermaine.francoeur\.vscode\extensions\ms-mssql.mssql-1.16.0\sqltoolsservice\4.2.1.5\Windows\MicrosoftSqlToolsServiceLayer.exe
    0	     2	 14216	       console-window-host (Windows internal process)
    0	   103	 12628	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=33648
    0	    16	 18192	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\dbaeumer.vscode-eslint-2.2.6\server\out\eslintServer.js --node-ipc --clientProcessId=33648
    0	   761	 32440	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0\server --logToConsole --ngProbeLocations c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas-v6-spa --includeAutomaticOptionalChainCompletions --includeCompletionsWithSnippetText --tsProbeLocations ,c:\Users\jermaine.francoeur\.vscode\extensions\angular.ng-template-14.2.0,c:\Users\jermaine.francoeur\repos\xcas-v6-spa --node-ipc --clientProcessId=33648
    0	    43	 33608	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --serverMode partialSemantic --useInferredProjectPerProjectRoot --disableAutomaticTypingAcquisition --cancellationPipeName C:\Users\JERMAI~1.FRA\AppData\Local\Temp\vscode-typescript\90c6ecbb0be4894a328a\tscancellation-cbac10e66e3275b4768e.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	   167	 37060	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node --max-old-space-size=3072 "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName C:\Users\JERMAI~1.FRA\AppData\Local\Temp\vscode-typescript\90c6ecbb0be4894a328a\tscancellation-87965bec307510e846a4.tmp* --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
    0	     2	 13512	       "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:/Users/jermaine.francoeur/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typingsInstaller.js" --globalTypingsCacheLocation C:/Users/jermaine.francoeur/AppData/Local/Microsoft/TypeScript/4.8 --enableTelemetry --typesMapLocation "c:/Users/jermaine.francoeur/AppData/Local/Programs/Microsoft VS Code/resources/app/extensions/node_modules/typescript/lib/typesMap.json" --validateDefaultNpmLocation
    0	    63	 44036	     "C:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\jermaine.francoeur\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\css-language-features\server\dist\node\cssServerMain" --node-ipc --clientProcessId=33648
    0	    14	 35624	   utility-network-service
    0	   135	 37408	   window (Contact.aspx.vb - private - Visual Studio Code)
    0	   222	 38068	   window (index.ts - xcas-v6-spa - Visual Studio Code)
    1	    86	 46672	   issue-reporter
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (Contact.aspx.vb - private - Visual Studio Code)
|  Window (index.ts - xcas-v6-spa - Visual Studio Code)
|  Window (index.ts - xcas-spa - Visual Studio Code)
|    Folder (private): 5362 files
|      File types: svg(1074) gif(680) png(434) jpg(384) asp(376) js(350)
|                  aspx(282) dll(277) vb(252) css(150)
|      Conf files: package.json(3) webpack.config.js(1) sln(1)
|    Folder (xcas-v6-spa): 6098 files
|      File types: json(2885) svg(1033) ts(456) png(256) scss(232) jpg(228)
|                  html(220) gif(212) pack(160) pdf(18)
|      Conf files: package.json(1) tsconfig.json(1)
|    Folder (xcas-spa): 1114 files
|      File types: json(885) pack(95) ts(34) html(11) scss(10) js(9) old(7)
|                  css(4) ico(3) txt(2)
|      Conf files: launch.json(1) tasks.json(1) package.json(1)
|                  tsconfig.json(1)
|      Launch Configs: pwa-chrome chrome;
```

</details>
<details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
csharp-to-typescript|adr|1.12.1
ng-template|Ang|14.2.0
vscode-eslint|dba|2.2.6
gitlens|eam|13.0.3
godot-tools|gee|1.3.1
git-graph|mhu|1.30.0
csharp|ms-|1.25.0
data-workspace-vscode|ms-|0.3.0
mssql|ms-|1.16.0
sql-bindings-vscode|ms-|0.3.0
sql-database-projects-vscode|ms-|0.19.0
vscode-thunder-client|ran|1.20.1
partial-diff|ryu|1.4.3

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
2144e591:30553903
pyindex848:30577860
nodejswelcome1:30587005
40g7c324:30573242
vscutilityc:30590830

```

</details>

**Comment by jflifecycle** [comment]
This is likely NOT the same root cause because as I mentioned, this occurs regardless of the number of files, or size of project. It is also not related to file size. The workspace details provided do not well reflect the issue reported, because it occurred in a Vue project, not an aspx project. That's just because I reported it on my work laptop.

This occurs because VSCode locks the file because it is using it and does not release it. It is not because of the number of watchers. This occurred in a small project with a small Typescript file. 

This has also been an issue for years, and as I stated, is possibly related to extensions. Which is why **I proposed a setting to ignore ANY checks before a file move.** Again, VsCode reported, without crashing, that it had failed to move some files. I was able to move other files in the project, just not the files I wanted to move. Because VsCode was using them and wouldn't let them go.

The CPU usage did not jump, it was just waiting to fail.

The reason I suggest the setting is because trying to reverse engineer which of the 30 extensions I use is a waste of my time, I'm just trying to move some files and VsCode is blocking me.
